eclair-core/src/main/scala/fr/acinq/eclair/payment/Bolt12Invoice.scala

```diff
@@ -145,27 +146,38 @@ object Bolt12Invoice {
    * @param preimage the preimage to use for the payment
    * @param nodeKey  the key that was used to generate the offer, may be different from our public nodeId if we're hiding behind a blinded route
    * @param features invoice features
+   * @param selfId   data to identify the payment when receiving it, can contain anything and is not readable by the payer
    */
-  def apply(offer: Offer, request: InvoiceRequest, preimage: ByteVector32, nodeKey: PrivateKey, features: Features[InvoiceFeature]): Bolt12Invoice = {
+  def apply(offer: Offer,
+            request: InvoiceRequest,
+            preimage: ByteVector32,
+            nodeKey: PrivateKey,
+            minFinalCltvExpiryDelta: CltvExpiryDelta,
+            features: Features[InvoiceFeature],
+            selfId: ByteVector): Bolt12Invoice = {
     require(request.amount.nonEmpty || offer.amount.nonEmpty)
+    val amount = request.amount.orElse(offer.amount.map(_ * request.quantity)).get
     val tlvs: Seq[InvoiceTlv] = Seq(
       Some(Chain(request.chain)),
-      Some(CreatedAt(TimestampSecond.now())),
-      Some(PaymentHash(Crypto.sha256(preimage))),
       Some(OfferId(offer.offerId)),
-      Some(NodeId(nodeKey.publicKey)),
-      Some(Amount(request.amount.orElse(offer.amount.map(_ * request.quantity)).get)),
+      Some(Amount(amount)),
       Some(Description(offer.description)),
+      if (!features.isEmpty) Some(FeaturesTlv(features.unscoped())) else None,
+      Some(Paths(Seq(Sphinx.RouteBlinding.createDirect(randomKey(), nodeKey.publicKey, selfId)))),
```

Comment:
Does it really make sense to include a 1-hop dummy blinded path (that's what we're doing here, right?)?
I think the blinded paths should be provided as arguments to this constructor instead, because they will be chosen and created by another component, won't they?

It feels to me that the flow will be:

- we receive an invoice request for an offer we created
- we extract an amount and use the graph to find suitable blinded paths
- we create the corresponding bolt 12 invoice
- we send it back to the requestor

Does that look like what you had in mind?

Reply:
Made `path` an argument.